tests/console/zypper_lifecycle_toolchain.pm

```diff
@@ -0,0 +1,52 @@
+# Copyright (C) 2017 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: test for 'zypper lifecycle' for toolchain module
+# Maintainer: Rodion Iafarov <riafarov@suse.com>
+# Tags: fate#32205
+
+use base "consoletest";
+use strict;
+use testapi;
+
+sub run() {
+    my $expected_date = '2024-10-30';
+    my $expired_date  = 'Now';
+
+    select_console 'root-console';
+    # Get gcc packages, ignore conflicting gcc6-ada and libada6 packages
+    my $gcc_packages = script_output
+      'zypper se -tpackage -r SLE-Module-Toolchain12-Updates | awk -F \'|\' \'NF > 1 {print $2}\' | tail -n +2 | fgrep -v gcc6-ada | fgrep -v libada6', 300;
```

Comment:
We could have reused that for the installation, but we still need a list of packages. And if we store packages list we also could use that for installation. As of now I reuse cmd from utils module.